supervisor/api/docker.py

```diff
@@ -0,0 +1,44 @@
+"""Init file for Supervisor Home Assistant RESTful API."""
+import logging
+from typing import Any, Dict
+
+from aiohttp import web
+
+from ..const import ATTR_HOSTNAME, ATTR_REGISTRIES, ATTR_USERNAME
+from ..coresys import CoreSysAttributes
+from ..docker.validate import SCHEMA_DOCKER_REGISTRY
+from .utils import api_process, api_validate
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+
+class APIDocker(CoreSysAttributes):
+    """Handle RESTful API for Docker configuration."""
+
+    @api_process
+    async def registries(self, request) -> Dict[str, Any]:
+        """Return the list of registries."""
+        data_registries = {}
+        for hostname, registry in self.sys_docker.config.registries.items():
+            data_registries[hostname] = {
+                ATTR_USERNAME: registry[ATTR_USERNAME],
+            }
+
+        return {ATTR_REGISTRIES: data_registries}
+
+    @api_process
+    async def create_registry(self, request: web.Request):
+        """Create a new docker registry."""
+        body = await api_validate(SCHEMA_DOCKER_REGISTRY, request)
```

Comment:
Define the schema in that file, because the file or API can change over time and go forget where it was used. You can look to other API parts how they do it.

Reply:
https://github.com/home-assistant/supervisor/blob/dev/supervisor/api/supervisor.py#L55-L70